### DISCUSSION TITLE
How can I add a timestamp before the note title to upload ?

### DISCUSSION THREAD

**Discussion by mathisgauthey** [ObsidianPublisher/obsidian-github-publisher#179] [Q&A — Plugin] [answered]
Hi there, I'm using the [following template](https://mmistakes.github.io/minimal-mistakes/docs/posts/) from Jekyll and just discovered this plugin that solves the main problem I had with obsidian to share things to the world.

However, I need to change my note title to "YYYY-MM-DD-title.md", but I don't want to change it inside obsidian for quality of life purpose.

Is this possible to add a date to the uploaded file in the plugin options ? 

That's about the only thing I need and everything works so well, I'll write something about how to do it, I struggled so much and so many people want a simple sharing workflow like that.

Thanks in advance !

**Comment by Mara-Li** [comment]
-> [In the documentation](https://obsidian-publisher.netlify.app/plugin/settings/upload/#filename)

**Reply by mathisgauthey** [reply]
Oh, I've seen that, issue is that I don't know anything about regex yet, I thought there was another way ! Time to learn then, thanks for the plugin, I'll make sure to share the word when I get something up and running !

**Reply by Mara-Li** [reply]
I don't think it's possible to automatically create a date. The best way is to use the `frontmatter` settings.

**Reply by mathisgauthey** [reply]
I don't really understand what you mean by that, but I guess my issue resolve around not using posts (blog posts type) inside Jekyll but using another collection of notes.

I'll have to tweak a bit cause I've never done anything like it. 

But it'll allow me not to have to use a timestamp and just use the export capabilities of your plugin !

**Reply by Mara-Li** [reply]
You have an option to change the filename of a note using the frontmatter. For example:
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/30244939/c36f7581-ac5e-4910-a507-173ba2292442)
(sorry for french)
I set "name" for that.

In the frontmatter, I set the `name` key, and set the filename I want. For example, `name: 01-01-01-note` will create the note `01-01-01-note.md`.

**Reply by mathisgauthey** [reply]
Ah alright yes I saw that on the blog that linked me to find your plugin, he used a filename frontmatter like that. But I'm going to stick on using the non-blog format from Jekyll, I just need to figure out how it works, but It'll prevent me from using additional timestamp as I already have frontmatter one's from obsidian !

And no worries, I'm a baguette 🥖 Merci encore 👌

**Reply by Mara-Li** [reply]
En soit, tu peux utiliser templater ou linter pour ajouter automatiquement la date, surtout si tu as déjà une clé frontmatter avec cette dernière.

**Reply by mathisgauthey** [reply]
Le soucis c'est que j'ai déjà les clefs suivantes :
```
---
title: Test
date created: 2023-07-16 18:09:21
date modified: 2023-07-16 18:10:35
tags: 
aliases: 
---
```

Je peux utiliser templater ou quoi mais je préfère ne pas toucher au nom de fichier, en fait j'aimerais ne pas utiliser le dossier _posts de Jekyll. J'ai vu qu'on pouvait utiliser des collections ou des pages directement, j'ai réussis à me dépatouiller là et ça fonctionne en local, plus qu'a voir avec le theme en remote et j'aurais une solution qui me fait pas changer mes clés frontmatter.

**Reply by mathisgauthey** [reply, marked as answer]
I ended up using Hugo for another static site generation that doesn't need a timestamp.

Im struggling with text replacement not working thought.

I set up this input : `//(\[\[([^\]]+)\]\])//`
And this output : `//[$2]({{< ref "$2" >}})//`

But my links are not converted while uploading, I don't understand why.

Edit : I suspect a bug somewhere in the replace text, my regex should be good.

![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/46576952/cbed41d7-372c-4e18-a9e7-4f538efc827a)

Edit : **The issue was the number of `//`, as stated I should enclose the text with "//", but in the end it's only one of them, not two slashs.**

Working solution : 
- input : `/\[\[([^\]]+)\]\]/`
- output : `[$1]({{< ref "$1" >}})`

But now I'm facing issues with attachments, it's never ending.

**Reply by Mara-Li** [reply]
What the problem with attachment?

**Reply by mathisgauthey** [reply]
No problem with the vault, I simple added a conversion regex to make the smooth transition from Obsidian ![[image.png]] to my hugo setup. Same for [[test]] to [test]({{< ref test >}}).

Now everything is fine tunned, I'll write some kind of guide if you want to create a Hugo template of something.

**Reply by Mara-Li** [reply]
Happy to know then! 💕
